### ISSUE TITLE
[warning] failed to set TCP_FASTOPEN:Protocol not available on FreeBSD 11.1-RELEASE-P9

### ISSUE THREAD

**Issue by ksaltik** [h2o/h2o#1732]
I get this warning
[warning] failed to set TCP_FASTOPEN:Protocol not available on FreeBSD 11.1-RELEASE-P9

Is H2o web server production ready ?.........It seems it changes very fast than other web server softwares.

**Comment by utrenkner** [comment]
My understanding of this: h2o warns you that your FreeBSD system does not support TCP Fast Open. Which would be the case, if you use the GENERIC 11.1 kernel (see the [generic kernel config of 11-STABLE](https://svnweb.freebsd.org/base/stable/11/sys/amd64/conf/GENERIC?revision=332040&view=markup)).

If you want to use TCP Fast Open, compile a new kernel with 
```
options TCP_RFC7413 
```
Note: I have not used TCP Fast Open yet..

**Comment by ksaltik** [comment]
Thank you for your answer. I wonder if the FreeBSD H2o package is mruby enabled

**Comment by utrenkner** [comment]
Yes, mruby is on. Easiest way to check is via [freshports](https://www.freshports.org/www/h2o/).
```
Configuration Options

===> The following configuration options are available for h2o-2.2.4:
     DOCS=on: Build and/or install documentation
     MRUBY=on: Build with embedded mruby handler support
===> Use 'make config' to modify these settings
```

**Comment by myfreeweb** [comment]
Info update: These days, you could instead see

> [warning] failed to set TCP_FASTOPEN:**Operation not permitted**

because server-side fast open is not enabled by default. To enable, run as root:

```
sysctl net.inet.tcp.fastopen.server_enable=1
```

and add

```
net.inet.tcp.fastopen.server_enable=1
```

to `/etc/sysctl.conf` to automatically enable on boot.
